clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -5169,6 +5171,72 @@ public void testShouldRefreshCoordinator() {
         assertTrue(ConsumerGroupOperationContext.shouldRefreshCoordinator(errorCounts));
     }
 
+    @Test
+    public void testDecommissionBrokerSuccess() throws InterruptedException, ExecutionException {
+        int decommissionedBrokerNode = 1;
+        try (AdminClientUnitTestEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(
+                    NodeApiVersions.create(ApiKeys.DECOMMISSION_BROKER.id, (short) 0, (short) 0));
+            env.kafkaClient().prepareResponse(prepareDecommissionBrokerResponse(Errors.NONE, 0));
+
+            DecommissionBrokerResult result = env.adminClient().decommissionBroker(decommissionedBrokerNode);
+
+            // Validate response
+            assertNotNull(result.all());
+            result.all().get();
+        }
+    }
+
+    @Test
+    public void testDecommissionBrokerFailure() {
+        int decommissionedBrokerNode = 1;
+        try (AdminClientUnitTestEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(
+                    NodeApiVersions.create(ApiKeys.DECOMMISSION_BROKER.id, (short) 0, (short) 0));
+            env.kafkaClient().prepareResponse(prepareDecommissionBrokerResponse(Errors.BROKER_NOT_AVAILABLE, 0));
```

Comment:
Hmm.. I would actually expect BROKER_NOT_AVAILABLE to be retriable. Maybe we should use an error code like `UNKNOWN_SERVER_ERROR`.